### PR TITLE
P1 14 js request help page

### PR DESCRIPTION
**Pull request by mike-f50**

### CODE REVIEW

File: HelpMyStreetFE/HelpMyStreetFE/Repositories/RequestHelpRepository.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using HelpMyStreetFE.Models.RequestHelp;
+using System;
+
+namespace HelpMyStreetFE.Repositories
+{
+	public class RequestHelpRepository : IRequestHelpRepository
+	{
+		public void PersistForm(RequestHelpFormModel form)
+		{
+			//TODO: api call
```
Comment (durbster) [inline HelpMyStreetFE/HelpMyStreetFE/Repositories/RequestHelpRepository.cs:10]: Does this TODO need addressing here?

File: HelpMyStreetFE/HelpMyStreetFE/src/js/requesthelp/requesthelp.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+		const postcode = $("input[name=postcode_search]").val();
+
+		try {
+			const resp = await fetch(`/api/postcode/checkCoverage/${postcode}`);
+			console.log(resp);
+			if (resp.ok) {						
+				const response = await resp.json();
+			
+				console.log(response);
```
Comment (durbster) [inline HelpMyStreetFE/HelpMyStreetFE/src/js/requesthelp/requesthelp.js:25]: log will need removing

File: HelpMyStreetFE/HelpMyStreetFE/Models/RequestHelp/RequestHelpFormModel.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        public string Email { get; set; }
+
+        [BindProperty(Name = "phonenumber")]
+        public string PhoneNumber { get; set; }
+
+        [MaxLength(500)]
+        [BindProperty(Name = "message")]
+        public string Message { get; set; }
+
+        [BindProperty(Name = "helpNeededArray")]
```
Comment (durbster) [inline HelpMyStreetFE/HelpMyStreetFE/Models/RequestHelp/RequestHelpFormModel.cs:33]: Should this be `help_needed_array` for consistency?

File: HelpMyStreetFE/HelpMyStreetFE/src/js/requesthelp/requesthelp.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+				"Please enter a valid first line of your address",
+		});
+
+		let validForm = (valid && validHelpNeeded);
+
+		let emailValid = true;
+		let emailInput = $("input[name='email']");
+		let emailInputError = emailInput.find("~ .error");
+
+		const emailRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```
Comment (usmaanrazaq) [inline HelpMyStreetFE/HelpMyStreetFE/src/js/requesthelp/requesthelp.js:108]: Email Valid Regex test  can should be used inside the validateformData function. 
Take a look at stepone.js for the password field, thats an example of using a regex match.
image

**Comment by mike-f50** [comment]
@jansevcik just triggering a notification as I raised the PR on Jan's behalf.

File: HelpMyStreetFE/HelpMyStreetFE/Repositories/RequestHelpRepository.cs (hunk 1 above)
Comment (jansevcik) [reply HelpMyStreetFE/HelpMyStreetFE/Repositories/RequestHelpRepository.cs:10]: I think we should replace TODO by connection to Request Service before we merge

Comment (jansevcik) [reply HelpMyStreetFE/HelpMyStreetFE/Repositories/RequestHelpRepository.cs:10]: Well, no, we can merge sooner and develop this backend later, as we want to put the form to dev server asap to gather initial feedback?

**Comment by mike-f50** [comment]
@usmaanrazaq are you happy now?

**Comment by jansevcik** [comment]
Please give me time till 3pm, I will spend another 2hrs to develop/review against specs.

**Comment by jansevcik** [comment]
Hi, please review my changes. If ok, please merge and deploy.

**Comment by usmaanrazaq** [comment]
Changes look good 👍

**Pull request merged**